tests/testautotls_integration.nim

```diff
@@ -32,105 +32,120 @@ import ./helpers
-      .withNoise()
-      .build()
-
-    # this is to quickly renew cert for testing
-    switch.autotls.renewCheckTime = 1.seconds
-
-    await switch.start()
-    defer:
-      await switch.stop()
-
-    # wait for cert to be ready
-    await switch.autotls.certReady.wait()
-    # clear since we'll use it again for renewal
-    switch.autotls.certReady.clear()
-
-    let dnsResolver = DnsResolver.new(DefaultDnsServers)
-    let base36PeerId = encodePeerId(switch.peerInfo.peerId)
-    let dnsTXTRecord = (
-      await dnsResolver.resolveTxt(
-        "_acme-challenge." & base36PeerId & "." & AutoTLSDNSServer
+  suite "AutoTLS Integration":
+    asyncTeardown:
+      checkTrackers()
+
+    asyncTest "request challenge without ACMEClient (ACMEApi only)":
+      let key = KeyPair.random(PKScheme.RSA, newRng()[]).get()
+      let acmeApi = ACMEApi.new(acmeServerURL = parseUri(LetsEncryptURLStaging))
+      defer:
+        await acmeApi.close()
+      let registerResponse = await acmeApi.requestRegister(key)
+      # account was registered (kid set)
+      check registerResponse.kid != ""
+      if registerResponse.kid == "":
+        raiseAssert "unable to register acme account"
+
+      # challenge requested
+      let challenge = await acmeApi.requestChallenge(
+        @["some.dummy.domain.com"], key, registerResponse.kid
       )
-    )[0]
+      check challenge.finalize.len > 0
+      check challenge.order.len > 0
 
-    # check if DNS TXT record is set
-    check dnsTXTRecord.len > 0
+      check challenge.dns01.url.len > 0
+      check challenge.dns01.`type` == ACMEChallengeType.DNS01
+      check challenge.dns01.status == ACMEChallengeStatus.PENDING
+      check challenge.dns01.token.len > 0
 
-    # certificate was downloaded and parsed
-    let cert = switch.autotls.cert.valueOr:
-      raiseAssert "certificate not found"
-    let certBefore = cert
-
-    # invalidate certificate
-    switch.autotls.certExpiry = Opt.some(Moment.now - 2.hours)
-
-    # wait for cert to be renewed
-    await switch.autotls.certReady.wait()
-
-    # certificate was indeed renewed
-    let certAfter = switch.autotls.cert.valueOr:
-      raiseAssert "certificate not found"
-
-    check certBefore != certAfter
-
-    let certExpiry = switch.autotls.certExpiry.valueOr:
-      raiseAssert "certificate expiry not found"
-
-    # cert is valid
-    check certExpiry > Moment.now
+    asyncTest "request challenge with ACMEClient":
+      let acme = ACMEClient.new(
+        api = ACMEApi.new(acmeServerURL = parseUri(LetsEncryptURLStaging))
+      )
+      defer:
+        await acme.close()
+
+      let challenge = await acme.getChallenge(@["some.dummy.domain.com"])
+
+      check challenge.finalize.len > 0
+      check challenge.order.len > 0
+      check challenge.dns01.url.len > 0
+      check challenge.dns01.`type` == ACMEChallengeType.DNS01
+      check challenge.dns01.status == ACMEChallengeStatus.PENDING
+      check challenge.dns01.token.len > 0
```

Comment:
you can also do something like this to avoid repeating `check`

```suggestion
      check:
        challenge.finalize.len > 0
        challenge.order.len > 0
        challenge.dns01.url.len > 0
        challenge.dns01.`type` == ACMEChallengeType.DNS01
        challenge.dns01.status == ACMEChallengeStatus.PENDING
        challenge.dns01.token.len > 0
```